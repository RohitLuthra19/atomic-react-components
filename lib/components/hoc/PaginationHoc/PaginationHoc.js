// @flow
/**
 *HOC
 * paginationHoc
 *
 */

import React, { useState, useEffect } from 'react';

import { getDisplayName } from '../../../core/utils/getDisplayName';
import Pagination from '../../molecules/Pagination';

const PaginationHocComponent = (WrappedComponent, data) => {
  // ...and returns another component...

  function PaginationHoc(props) {
    const [posts, setPosts] = useState([]);
    const [currentPage, setCurrentPage] = useState(0);
    const [postsPerPage] = useState(12);
    const [currentData, setCurrentData] = useState([]);

    useEffect(() => {
      setPosts(data.items);
    }, []);

    const indexOfLastPost = currentPage * postsPerPage;
    const indexOfFirstPost = indexOfLastPost - postsPerPage;

    useEffect(() => {
      setCurrentPage(1);
    }, []);

    useEffect(() => {
      const updatedData =
        posts && posts.length > 0 && posts.slice(indexOfFirstPost, indexOfLastPost);
      setCurrentData(updatedData);
    }, [currentPage]);

    const paginate = pageNumber => setCurrentPage(pageNumber);

    return (
      <>
        <WrappedComponent {...props}>{paginationContent(currentData)}</WrappedComponent>
        <Pagination
          postsPerPage={postsPerPage}
          totalPosts={posts.length}
          paginate={paginate}
          currentPage={currentPage}
          firstPost={indexOfFirstPost}
          lastPost={indexOfLastPost}
        />
      </>
    );
  }

  PaginationHoc.displayName = `PaginationHoc(${getDisplayName(WrappedComponent)})`;
  return PaginationHoc;
};

export default PaginationHocComponent;
export { PaginationHocComponent as PaginationHocVanilla };
