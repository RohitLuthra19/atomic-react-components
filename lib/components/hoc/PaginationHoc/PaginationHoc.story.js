import React from 'react';
import { storiesOf } from '@storybook/react';
// import { action } from '@storybook/addon-actions';
import defaultConfig from './PaginationHoc.mock';

// Import Styled Component to showcase variations
import PaginationHocComponent, { PaginationHocVanilla } from '.';
import List from '../../molecules/List';

// component with render prop

const PaginationWithRenderProp = paginationContent => (
  <List>
    {paginationContent => (
      <div>
        <h1>Vishwesh</h1>
      </div>
    )}
  </List>
);

const PaginationList = PaginationHocComponent(<PaginationWithRenderProp />, { ...defaultConfig });
const PaginationListVanilla = PaginationHocVanilla(List, { ...defaultConfig });

storiesOf('HOC/PaginationHoc', module)
  .addParameters({ jest: ['PaginationHoc', 'PaginationHocVanilla'] })
  .add('Pagination', () => <PaginationListVanilla />)
  .add('PaginationHoc', () => <PaginationList />);
