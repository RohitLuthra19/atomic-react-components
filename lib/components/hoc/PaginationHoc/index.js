// @flow
export { default } from './PaginationHoc';
export { PaginationHocVanilla } from './PaginationHoc';
