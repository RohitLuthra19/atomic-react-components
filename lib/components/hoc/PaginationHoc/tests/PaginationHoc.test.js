import React from 'react';
import { shallow, render, mount } from 'enzyme';

import PaginationHoc, { PaginationHocVanilla } from '../index';

describe('<PaginationHoc />', () => {
  let PaginationHocComponent = '';
  beforeEach(() => {
    PaginationHocComponent = shallow(<PaginationHocVanilla>Test</PaginationHocVanilla>);
  });

  test('should render correctly', () => {
    expect(PaginationHocComponent).toMatchSnapshot();
  });
});
