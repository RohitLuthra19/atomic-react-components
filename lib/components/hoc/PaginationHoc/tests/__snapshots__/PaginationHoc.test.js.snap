// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaginationHoc /> should render correctly 1`] = `<Component />`;

exports[`<PaginationHoc /> should render correctly 2`] = `<Component />`;
